tests/algorithms/test_algorithm_resumption.py

```diff
@@ -1,105 +1,106 @@
 # Copyright 2022 MosaicML Composer authors
 # SPDX-License-Identifier: Apache-2.0
 
+import copy
 import os
 import pathlib
-from typing import Optional
+from typing import Any, Callable, Dict
 
 import pytest
 import torch
-from torch.utils.data import DataLoader
+from torch.utils.data import DataLoader, Dataset
 
 from composer import Trainer
-from composer.algorithms import get_algorithm_registry
-from tests.algorithms.algorithm_settings import get_settings
+from composer.algorithms.factorize.factorize import Factorize
+from composer.algorithms.layer_freezing.layer_freezing import LayerFreezing
+from composer.algorithms.sam.sam import SAM
+from composer.algorithms.squeeze_excite.squeeze_excite import SqueezeExcite
+from composer.algorithms.stochastic_depth.stochastic_depth import StochasticDepth
+from composer.core.algorithm import Algorithm
+from composer.models.base import ComposerModel
+from tests.algorithms.algorithm_settings import get_algorithm_parametrization
 from tests.common import deep_compare, device
 
-ALGORITHMS = get_algorithm_registry().keys()
+
+@pytest.fixture
+def model(request):
+    # Copy the model, so other parameterization start with a fresh model
+    return copy.deepcopy(request.param)
 
 
 @pytest.mark.timeout(180)
 @device('gpu')
 @pytest.mark.parametrize(
-    "seed,save_interval,save_filename,resume_file,final_checkpoint",
+    "save_interval,save_filename,resume_file,final_checkpoint",
     [
-        [None, "1ep", "ep{epoch}-rank{rank}", "ep2-rank{rank}", "latest-rank{rank}"
-        ],  # test randomized seed saving and symlinking
-        [42, "1ep", "ep{epoch}-rank{rank}", "ep3-rank{rank}", "ep5-rank{rank}"],  # test save at epoch end
+        ["1ep", "ep{epoch}-rank{rank}", "ep2-rank{rank}", "latest-rank{rank}"],  # symlinking
+        ["1ep", "ep{epoch}-rank{rank}", "ep3-rank{rank}", "ep5-rank{rank}"],  # test save at epoch end
     ],
 )
-@pytest.mark.filterwarnings(r"ignore:.*Detected call of \`lr_schedule.*:UserWarning")  # Needed for SWA
-@pytest.mark.parametrize("algorithm", ALGORITHMS)
+@pytest.mark.parametrize("alg_cls,alg_kwargs,model,dataset", get_algorithm_parametrization(), indirect=['model'])
 def test_algorithm_resumption(
-    algorithm: str,
     device,
-    seed: Optional[int],
     save_interval: int,
     save_filename: str,
     resume_file: str,
     final_checkpoint: str,
     tmp_path: pathlib.Path,
+    alg_cls: Callable[..., Algorithm],
```

Comment:
`Type[Algorithm]`?

Reply:
If typed as `Type[Algorithm]`, then pyright complains that the base `Algorithm` class takes no arguments. I think I'd need to use a `TypeVar(bound=Algorithm)` to get around that, so would prefer to leave it as a callable.

Reply:
I added in a stub init signature into the parent class for `Algorithm`, so then `Type[Algorithm]` would work.